### ISSUE TITLE
Buildkite not listed as an OIDC provider in CLI docs

### ISSUE THREAD

**Issue by tylerauerbeck** [sigstore/cosign#2921]
**Description**

When attempting to use the new Buildkite OIDC provider in the latest release, I do not see it listed as an option

```
--oidc-provider string                                                                     [EXPERIMENTAL] Specify the provider to get the OIDC token from (Optional). If unset, all options will be tried. Options include: [spiffe, google, github, filesystem]
```

Additionally, when attempting to set this (`--oidc-provider=buildkite`), I receive the following error:

```
Error: signing [quay.io/equinixmetal/kitewrangler:81-526c5a7b]: getting signer: getting key from Fulcio: getting provider: buildkite is not a valid provider
```

This turns out to be because the oidc-provider is registered as `buildkite-agent`: https://github.com/sigstore/cosign/blob/main/pkg/providers/buildkite/buildkite.go#L29-L31

This makes sense from a BK/implementation perspective, but was a bit of a surprise/misleading due to the missing docs.

** This all works magically and as expected when you're not supplying any of the additional flags due to the ambient OIDC bit, so it's really only a concern when you're specifying values via the CLI flags.

**Comment by hectorj2f** [comment]
@tylerauerbeck Yes, I agree this should recognize `buildkite` as a provider name.

**Comment by tylerauerbeck** [comment]
I think the most important thing is that we update the cli docs with _something_. I don't have too strong of an opinion on whether that's Buildkite vs Buildkite-agent. But either way, happy to PR changes once it's decided what that is.
